### PR TITLE
Fix nightly builds

### PR DESCRIPTION
**Pull request by alexflorisca**
## What
Fixes the nightly builds Github workflow. I've increased the node heap size limit to allow for the extra power the build script needs recently. I've created an issue (https://github.com/woocommerce/woocommerce/issues/42244) to monitor this over time so we can avoid this in the future, although I suspect we're running into problems here because we're generating a development build here, not production, which has a lot more bells and whistles.

## Why

The nightly build workflow is failing because the build process is using up too much memory (p1692008212605009-slack-C02UBB1EPEF)

## Testing Instructions

N/A

## Developer Testing Instructions

1. Go to https://github.com/woocommerce/woocommerce-blocks/actions/workflows/nightly-builds.yml
2. Click `Run Workflow` button.
3. Choose `fix/nightly-builds` as the branch
4. Click `Run Workflow`.
5. The workflow should complete successfully

* [x] Do not include in the Testing Notes 
* [ ] Should be tested by the development team exclusively 

## Screenshots or screencast

N/A

## WooCommerce Visibility

Required:

* [x] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [ ] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [x] This PR is assigned to a milestone.

Conditional:
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

N/A

### CODE REVIEW

File: .github/workflows/nightly-builds.yml [hunk 1]
```diff
@@ -5,6 +5,9 @@ on:
     - cron: '0 0 * * *' # Run at 12 AM UTC.
   workflow_dispatch:
 
+env:
+  NODE_OPTIONS: --max-old-space-size=4096 # development build takes a lot of memory 
+
```
Comment (senadir) [inline .github/workflows/nightly-builds.yml:10]: This should give us plenty of space, I presume the old one was 2048?

Comment (senadir) [reply .github/workflows/nightly-builds.yml:10]: I don't suspect we will hit this limit any time soon, hitting it means our local build is around 2mns which is going to significant increase.

**Review by senadir — COMMENTED** [review]
Looks good to me.

**Review by senadir — APPROVED** [review]
Approve didn't go first time!

File: .github/workflows/nightly-builds.yml (hunk 1 above)
Comment (alexflorisca) [reply .github/workflows/nightly-builds.yml:10]: > This should give us plenty of space, I presume the old one was 2048?

I'm not sure actually, I thought node 16 uses 4gb by default, but I think it depends on the machine it runs on as well. If the github workers only have 4gb. maybe it sets it lower

Comment (alexflorisca) [reply .github/workflows/nightly-builds.yml:10]: > I don't suspect we will hit this limit any time soon, hitting it means our local build is around 2mns which is going to significant increase.

Yeah also bare in mind this is for the dev build not prod build, so we shouldn't run into this on other actions

**Pull request merged**